frontend/appflowy_flutter/lib/workspace/presentation/home/panes/draggable_pane_target.dart

```diff
@@ -0,0 +1,238 @@
+}
+
+class _DraggablePaneTargetState extends State<DraggablePaneTarget> {
+  FlowyDraggableHoverPosition position = FlowyDraggableHoverPosition.none;
+  @override
+  Widget build(BuildContext context) {
+    return DraggableItemTarget<CrossDraggablesEntity>(
+      onWillAccept: (data) => _shouldAccept(data!, position),
+      onMove: (data) {
+        final renderBox = widget.paneContext.findRenderObject() as RenderBox;
+        final offset = renderBox.globalToLocal(data.offset);
+        final positionN = _computeHoverPosition(
+          offset,
+          renderBox,
+          data.data.crossDraggableType,
+        );
+        if (!_shouldAccept(data.data, position)) {
+          return;
+        }
+
+        setState(() {
+          position = positionN;
+        });
+      },
+      onLeave: (_) => setState(() {
+        position = FlowyDraggableHoverPosition.none;
+      }),
+      onAccept: (data) {
+        _move(data, widget.pane);
+        setState(() {
+          position = FlowyDraggableHoverPosition.none;
+        });
+      },
+      child: Stack(
+        children: [
+          widget.child,
+          _buildChildren(widget.paneContext, position),
+        ],
+      ),
+    );
+  }
+
+  Widget _buildChildren(
+    BuildContext context,
+    FlowyDraggableHoverPosition position,
+  ) {
+    final top = (widget.pane.draggable as PaneNode).tabs.pages > 1
+        ? HomeSizes.tabBarHeigth + HomeSizes.topBarHeight
+        : HomeSizes.topBarHeight;
+    return switch (position) {
+      FlowyDraggableHoverPosition.top => Positioned(
+          top: top,
+          child: Container(
+            height: (widget.size.height) / 2,
+            width: widget.size.width,
+            color: Theme.of(context).hoverColor.withOpacity(0.5),
+          ),
+        ),
+      FlowyDraggableHoverPosition.left => Positioned(
+          left: 0,
+          top: top,
+          child: Container(
+            width: widget.size.width / 2,
+            height: widget.size.height,
+            color: Theme.of(context).hoverColor.withOpacity(0.5),
+          ),
+        ),
+      FlowyDraggableHoverPosition.right => Positioned(
+          top: top,
+          left: widget.size.width / 2,
+          child: Container(
+            width: widget.size.width / 2,
+            height: widget.size.height,
+            color: Theme.of(context).hoverColor.withOpacity(0.5),
+          ),
+        ),
+      FlowyDraggableHoverPosition.bottom => Positioned(
+          top: widget.size.height / 2,
+          child: Container(
+            height: widget.size.height / 2,
+            width: widget.size.width,
+            color: Theme.of(context).hoverColor.withOpacity(0.5),
+          ),
+        ),
+      FlowyDraggableHoverPosition.tab => Positioned(
+          top: 0,
+          child: Container(
+            height: top,
+            width: widget.size.width,
+            color: Theme.of(context).hoverColor.withOpacity(0.5),
+          ),
+        ),
```

Comment:
Can we rewrite this to only calculate `top, left, right, bottom`, so that we only need one Widget.

Maybe something like this?
```
(
  double? left,
  double? top,
  double? right,
  double? bottom,
) _getPositioned(FlowyDraggableHoverPosition position) {
  double? left, top, right, bottom;

  switch (position) {
    ...
  }

  return (left, top, right, bottom);
}
```

Reply:
Oh yeah that would be better